File: back/src/main/java/com/altranapp/entities/ConsultantCompetence.java
package com.altranapp.entities;


import lombok.Data;

import javax.persistence.*;

@Entity
@Table(name = "Consultant_Competence")
@Data
public class ConsultantCompetence {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Consultant consultant;
    @ManyToOne
    private Competence competence;
    @ManyToOne
    private Niveau niveau;

}

File: front/ng-app-app-altran/src/app/consultant-competence/models/consultant-competence.ts
import { Consultant } from 'src/app/consultant/models/consultant';
import { Niveau } from 'src/app/niveau/models/niveau';
import { Competence } from 'src/app/competence/models/competence';

export class ConsultantCompetence {
    public constructor(
        public id: number,
        public consultant: Consultant,
        public competence: Competence,
        public niveau: Niveau
    ) { }

}
